internal/data/model/resource.go

```diff
@@ -0,0 +1,20 @@
+package model
+
+import (
+	"time"
+
+	"github.com/google/uuid"
+)
+
+type Resource struct {
+	ID               uuid.UUID `gorm:"type:uuid;primaryKey"`
+	Type             string    `gorm:"size:128;not null;"`
+	CommonVersion    uint      `gorm:"type:bigint;check:common_version >= 0"`
+	ConsistencyToken string    `gorm:"size:1024;column:ktn;"`
+	CreatedAt        time.Time
+	UpdatedAt        time.Time
+}
+
+func (Resource) TableName() string {
+	return "resource"
```

Comment:
Is the plan here that we will be operating under a singularly named `resource` table for the new schema for now and replacing the plural legacy `resources` at a later date?